### ISSUE TITLE
Bug in module abel.basex

### ISSUE THREAD

**Issue by bwohlberg** [PyAbel/PyAbel#321]
The following error occurs when running the example `examples/example_basex_step.py`:
```
(py38) ubuntu:PyAbel$ python examples/example_basex_step.py 
Traceback (most recent call last):
  File "examples/example_basex_step.py", line 30, in <module>
    recon_right = basex_transform(
  File "/tmp/PyAbel/build/lib/abel/basex.py", line 153, in basex_transform
    A = get_bs_cached(n, sigma=sigma, reg=reg, correction=correction,
  File "/tmp/PyAbel/build/lib/abel/basex.py", line 323, in get_bs_cached
    for f in listdir(basis_dir):
FileNotFoundError: [Errno 2] No such file or directory: 'bases'
```

This error occurs both with pyabel installed via pip and built from source.

**Comment by MikhailRyazanov** [comment]
As I understand, this “bug” is not in the `abel.basex` module but in the example. It seems that you are running it in a directory which does not have a `bases` subdirectory to store basis files. Try to go inside the `examples` directory first (it must already have the `bases` subdirectory) and run this example from there. Or create a `bases` subdirectory in the working directory where you want to run the code.

**Comment by bwohlberg** [comment]
Thanks for the explanation. I didn't notice the `basis_dir` parameter, and assumed that the examples could be run from the repo root. A README in the examples directory would be helpful.

**Comment by MikhailRyazanov** [comment]
Yes, the examples probably need some more documentation...

Another thing that could be improved is to check whether the `bases` directory exists, and if not, offer to run (or just complain and run) without storing the basis. This was addressed in #277 for the GUI example, where it was causing real problems, but can be done (more easily) for other examples as well.

**Comment by bwohlberg** [comment]
From a user perspective, the most useful behaviour would probably be for `abel.basex.basex_transform` to default to reading and writing from something like `$HOME/.cache/pyabel`.

**Comment by MikhailRyazanov** [comment]
This was also to some extent discussed [in #247](https://github.com/PyAbel/PyAbel/issues/247#issuecomment-474986998). :–) But we need to find a good portable way to do so.

**Comment by MikhailRyazanov** [comment]
@bwohlberg, please check PR #322 with a README file in the examples directory. Does it look fine, or you have some suggestions?

I'll probably open a separate issue to remind us about implementing default basis directories (separately from the GUI work).

**Comment by bwohlberg** [comment]
The README looks good.

One other suggestion: to make the script more robust to the directory from which is is run, use something like this to set the bases path:
```python
import sys
import os.path

bases = os.path.join(os.path.dirname(sys.argv[0]), 'bases')
```

**Comment by MikhailRyazanov** [comment]
Well, we need to ask @DanHickstein and @stggh about this, but my guess is that these examples are supposed to be simple and concise rather than robust...

**Comment by MikhailRyazanov** [comment]
I'm closing this now as at least partially addressed in #322. Will open another discussion about default basis directories and sample data later...
